service/src/main/java/greencity/service/HabitServiceImpl.java

```diff
@@ -676,6 +681,31 @@ public PageableDto<HabitDto> getAllFavoriteHabitsByLanguageCode(UserVO userVO, P
         return buildPageableDtoForDifferentParameters(habitTranslationPage, userVO.getId());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PageableDto<UserFriendHabitInviteDto> findAllFriendsOfUser(UserVO userVO, String name, Pageable pageable,
+        Long habitId) {
+        Long userId = userVO.getId();
+        name = Optional.ofNullable(name).orElse("");
+        PageableDto<UserFriendDto> userFriendDtoPageable = friendService.findAllFriendsOfUser(userId, name, pageable);
+        List<UserFriendHabitInviteDto> extendedDtoList = userFriendDtoPageable.getPage().stream()
+            .map(friend -> {
+                boolean hasInvitation = checkIfInvited(userId, friend.getId(), habitId);
```

Comment:
can we get a list of friends who have invitation instead of calling DB for every friend?